include/seqan/basic/allocator_interface.h

```diff
@@ -204,6 +204,7 @@ allocate(T const &,
     if (data)
         SEQAN_PROADD(SEQAN_PROMEMORY, count * sizeof(TValue));
 #endif
+	memset((void*)data, 0, count * sizeof(TValue));
```

Comment:
I think maybe some higher layers rely on it being initialized to 0 even if they should not. One way to test it, would be to do a memset 0xFF and see if the unit_tests on GCC and CLANG are still ok.